<launch>
    <arg name="base"       default="$(optenv TURTLEBOT_BASE kobuki)"/>  <!-- create, rhoomba -->
    <arg name="stacks"     default="$(optenv TURTLEBOT_STACKS hexagons)"/>  <!-- circles, hexagons -->
    <arg name="3d_sensor"  default="$(optenv TURTLEBOT_3D_SENSOR kinect)"/>  <!-- kinect, asus_xtion_pro -->
    <param name ="use_sim_time" value="true"/> 
    <node pkg="rosbag" type="play" name="player1" output="screen" args="--quiet --clock $(find dslam)/launch/dataset/intel.bag">
        <!--remap from="/submap_list" to = "/submap_list_1" />
        <remap from ="/imu" to = "imu_1" />
         <remap from ="/laser_scan" to = "/scan" /-->
        <!--remap from="/tf" to="/tf1"/ $(find dslam)/launch/testdata-2018-04-11-15-11-59.bag -->
        <!--remap from="/tf" to="/tf1"/-->
    </node>

   <include file="$(find turtlebot_bringup)/launch/includes/robot.launch.xml">
    <arg name="base" value="$(arg base)" />
    <arg name="stacks" value="$(arg stacks)" />
    <arg name="3d_sensor" value="$(arg 3d_sensor)" />
  </include>

   <node pkg="tf" type="static_transform_publisher" name="link1_broadcaster" args="0.110 0 0.02478 0 0 0 1 base_link laser 50" />

    <!--node pkg="distributed_slam_karto" type="d_slam_karto" name="d_slam_karto" output="screen">
        <param name="odom_frame" value="odom"/>
        <param name="map_update_interval" value="5"/>
        <param name="resolution" value="0.05"/>
        <rosparam command="load" file="$(find slam_karto_g2o)/config/mapping_karto_g2o.yaml" />
  </node-->
  
  <node pkg="distributed_slam_karto" type="d_slam_karto" name="d_slam_karto" output="screen"   >
    <param name="conf_file" value="$(find distributed_slam_karto)/launch/config.lua"/>
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find dslam)/launch/rvizconf.rviz"/>
</launch>